src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -66,7 +66,16 @@ public IList<IRemoteDependencyProvider> FilterDependencyProvidersForLibrary(Libr
             // filter package namespaces if enabled            
             if (PackageNamespaces?.AreNamespacesEnabled == true && libraryRange.TypeConstraintAllows(LibraryDependencyTarget.Package))
             {
-                IReadOnlyList<string> sources = PackageNamespaces.GetConfiguredPackageSources(libraryRange.Name);
+                IReadOnlyList<string> sources = default;
+
+                try
+                {
+                    sources = PackageNamespaces.GetConfiguredPackageSources(libraryRange.Name);
```

Comment:
I think I prefer if we validate outside instead of relying on an exception? 

Curious what others think. 

@erdembayar @zivkan